src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -304,17 +304,7 @@ async Task ExecuteInvocation()
                                 // And normal invocations handle cleanup below in the finally
                                 if (isStreamCall)
                                 {
-                                    hubActivator?.Release(hub);
-                                    scope.Dispose();
-                                    foreach (var stream in hubMethodInvocationMessage.StreamIds)
-                                    {
-                                        try
-                                        {
-                                            connection.StreamTracker.Complete(CompletionMessage.Empty(stream));
-                                        }
-                                        // ignore failures, it means the client already completed the streams
-                                        catch { }
-                                    }
+                                    CleanupInvocation(connection, hubMethodInvocationMessage, hubActivator, hub, scope);
```

Comment:
If you kept disposeScope true in the isStreamCall case, couldn't you just remove this call to CleanupInvocation() and let the call at https://github.com/aspnet/AspNetCore/pull/6655/files#diff-3736ce08aacd6dc93817f6f2fe3b0d1eR345 handle it?

Reply:
We need to delay in the upload stream case too.